overlord/snapstate/backend/link.go

```diff
@@ -26,12 +26,17 @@ import (
 
 	"github.com/snapcore/snapd/boot"
 	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/progress"
 	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/timings"
 	"github.com/snapcore/snapd/wrappers"
 )
 
+func isSnapFirstInstall(info *snap.Info) bool {
+	return osutil.IsSymlink(filepath.Join(info.MountDir(), "..", "current"))
```

Comment:
Right, that's a missing test case. Added the scenario and fixed the check, because `current` is removed during refresh, so we would incorrectly detect such scenario as first install.